### PR TITLE
Main development done!

### PR DESCRIPTION
**Pull request by stefanak-michal**

### CODE REVIEW

**Comment by stefanak-michal** [comment]
It still has to tested for frontend implementation. Add cypress or playwrigth automatic testing.

**Comment by stefanak-michal** [comment]
update docker to use https://github.com/orgs/thingsdb/discussions/366#discussioncomment-9056754

File: src/ThingsDB.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                    console.error('Message type check error');
+                    return;
+                }
+
+                let message: any = null;
+                if (size) {
+                    message = decode(event.data.slice(8));
+                }
+
+                if (id in this.pending) {
```
Comment (joente) [inline src/ThingsDB.ts]: Pending should only be checked for types 16, 17, 18 and 19. _(`16..31` is the range reserved for client responses)_
Other types are never a response to a request and therefore the "id" should ignored.

File: ci/thingsdb.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        libuv1-dev \
+        libpcre2-dev \
+        libyajl-dev \
+        libcurl4-nss-dev \
+        build-essential \
+        git \
+        libwebsockets-dev
+
+# Switch to the lws branch
+RUN git clone --branch lws https://github.com/thingsdb/ThingsDB.git ThingsDB
```
Comment (joente) [inline ci/thingsdb.Dockerfile:16]: The `lws` branch is no longer needed, `lws` is merged with the `main` branch.

**Pull request merged**